src/helpers/devserver.ts

```diff
@@ -0,0 +1,93 @@
+import fetch from "cross-fetch";
+
+const envVars = [
+  "NODE_ENV", // express slash standard.
+  "VERCEL_ENV", // vercel
+  "CONTEXT", // netlify
+];
+
+export const available = async (): Promise<boolean> => {
+  if (isProd()) {
+    return false;
+  }
+
+  try {
+    const url = devserverURL("/dev");
+    const result = await fetch(url.toString());
+    await result.json();
+    return true;
+  } catch (e) {}
+
+  return false;
+};
+
+// isProd compares any supported standard env variable for "production",
+// returning true on first match.
+export const isProd = (): boolean => {
+  return !!envVars.find((e) => process.env[e] === "production");
+};
+
+// url returns the dev server URL, overriding to use the INNGEST_DEVSERVER_URL
+// env var if provided.
+export const devserverURL = (pathname?: string): URL => {
+  let host = process.env.INNGEST_DEVSERVER_URL;
+  if (!host) {
+    // Use the default.
+    const url = new URL(`http://127.0.0.1:8288/`);
+    url.pathname = pathname || "";
+    return url;
+  }
+
+  // Normalize the URL to be friendly here.  If the user hasn't added a scheme,
+  // default to http.
+  if (host.indexOf("://") === -1) {
+    host = `http://${host}`;
+  }
+
+  const url = new URL(host);
+  url.pathname = pathname || "";
+  return url;
+};
```

Comment:
Nitpicking - can this be more succinct? I think this is the same functionality

```suggestion
export const devserverURL = (pathname = ""): URL => {
  let host =
    process.env.INNGEST_DEVSERVER_URL?.trim() || "http://127.0.0.1:8288/";
  if (!host.includes("://")) {
    host = `http://${host}`;
  }

  return new URL(pathname, host);
};
```